integration-tests/functions/__tests__/functions.js

```diff
@@ -0,0 +1,205 @@
+    const result = await fetch(
+      `http://localhost:8000/api/env-variables`
+    ).then(res => res.text())
+
+    expect(result).toEqual(`word`)
+  })
+})
+
+describe(`typescript`, () => {
+  test(`typescript functions work`, async () => {
+    const result = await fetch(
+      `http://localhost:8000/api/i-am-typescript`
+    ).then(res => res.text())
+
+    expect(result).toMatchSnapshot()
+  })
+})
+
+describe(`response formats`, () => {
+  test(`returns json correctly`, async () => {
+    const res = await fetch(`http://localhost:8000/api/i-am-json`)
+    const result = await res.json()
+
+    const { date, ...headers } = Object.fromEntries(res.headers)
+    expect(result).toMatchSnapshot()
+    expect(headers).toMatchSnapshot()
+  })
+  test(`returns text correctly`, async () => {
+    const res = await fetch(`http://localhost:8000/api/i-am-typescript`)
+    const result = await res.text()
+
+    const { date, ...headers } = Object.fromEntries(res.headers)
+    expect(result).toMatchSnapshot()
+    expect(headers).toMatchSnapshot()
+  })
+})
+
+describe(`functions can send custom statuses`, () => {
+  test(`can return 200 status`, async () => {
+    const res = await fetch(`http://localhost:8000/api/status`)
+
+    expect(res.status).toEqual(200)
+  })
+
+  test(`can return 404 status`, async () => {
+    const res = await fetch(`http://localhost:8000/api/status?code=404`)
+
+    expect(res.status).toEqual(404)
+  })
+
+  test(`can return 500 status`, async () => {
+    const res = await fetch(`http://localhost:8000/api/status?code=500`)
+
+    expect(res.status).toEqual(500)
+  })
+})
+
+describe(`functions can parse different ways of sending data`, () => {
+  test(`query string`, async () => {
+    const result = await fetch(
+      `http://localhost:8000/api/parser?amIReal=true`
+    ).then(res => res.json())
+
+    expect(result).toMatchSnapshot()
+  })
+
+  test(`form parameters`, async () => {
+    const { URLSearchParams } = require("url")
+    const params = new URLSearchParams()
+    params.append("a", `form parameters`)
+    const result = await fetch(`http://localhost:8000/api/parser`, {
+      method: `POST`,
+      body: params,
+    }).then(res => res.json())
+
+    expect(result).toMatchSnapshot()
+  })
+
+  test(`form data`, async () => {
+    const FormData = require("form-data")
+
+    const form = new FormData()
+    form.append("a", `form-data`)
+    const result = await fetch(`http://localhost:8000/api/parser`, {
+      method: `POST`,
+      body: form,
+    }).then(res => res.json())
+
+    expect(result).toMatchSnapshot()
+  })
+
+  test(`json body`, async () => {
+    const body = { a: `json` }
+    const result = await fetch(`http://localhost:8000/api/parser`, {
+      method: `POST`,
+      body: JSON.stringify(body),
+      headers: { "Content-Type": "application/json" },
+    }).then(res => res.json())
+
+    expect(result).toMatchSnapshot()
+  })
+
+  // test(`file in multipart/form`, async () => {
```

Comment:
Things commented out should be either removed or fixed and implemented :)

Reply:
I don't have strong feelings about this so can remove them if you'd like. I'm tracking the work to be done to make these test work [here](https://app.clubhouse.io/gatsbyjs/story/29330/be-able-to-handle-file-uploads) but this PR is just for the initial implementation of functions and adding file support will come in a follow-on PR.

Reply:
Yeah, fine, too 👍 The TODO comment gives context